sycl/source/detail/kernel_impl.hpp

```diff
@@ -185,6 +186,17 @@ class kernel_impl {
 
   const DeviceImageImplPtr &getDeviceImage() const { return MDeviceImageImpl; }
 
+  pi_native_handle getNative() const {
+    const plugin &Plugin = MContext->getPlugin();
+
+    Plugin.call<PiApiKind::piKernelRetain>(MKernel);
```

Comment:
I think similar to other SYCL objects "retain" should only be done for OpenCL-backend